packages/client/src/pages/planet/planet.contex.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import { getRequest } from '../../shared/services/http'
+import logger from 'use-reducer-logger'
+
+const PlanetContext = React.createContext()
+// status to handle status context
+export const PLANET_STATUS = {
+  INIT: 'INIT',
+  IDLE: 'IDLE',
+  GETTING: 'GETTING'
+}
+
+const initialState = {
+  status: PLANET_STATUS.INIT,
+  name: 'Earth',
+  message: {
+    planet: null
+  },
+  error: {
+    planet: null
+  }
+}
+
+// the action constants
+const actions = {
+  SET_PLANET: 'SET_PLANET'
+}
+
+// the state reducer switch
+export function planetReducer (state, action) {
+  switch (action.type) {
+    case actions.SET_PLANET:
+      return {
+        ...state,
+        status: PLANET_STATUS.IDLE,
+        name: action.payload,
+        message: {
+          planet: `You are now on planet ${action.payload}!`
+        },
+        error: {
+          planet: null
+        }
+      }
+
+    default:
+      break
+  }
+}
+
+// the reducer context with logging
```

Comment:
Im game to use this!
Though would have thought we could have used a library of some sort as it has it all baked in for us,

But either way I dont really care as long as it works :D

- would it be possible we made these into factory functions of some sort? So we dont need to write so much each time we make a new context?

Reply:
I would love to have some factory functions l! Lets do it! but I wouod prefer not having any state logic in libraries for security reasons